CSharpSourceCode/SubModule.cs

```diff
@@ -106,6 +107,8 @@ protected override void OnGameStart(Game game, IGameStarter gameStarterObject)
             else if(game.GameType is Campaign)
             {
                 CampaignGameStarter starter = gameStarterObject as CampaignGameStarter;
+                PartyAttributeManager partyAttributeManager = PartyAttributeManager.Instance;
+                starter.CampaignBehaviors.Add(partyAttributeManager);
```

Comment:
PartyAttributeManager is accessible through CampaignBehaviors. It's also a singleton. This seems a bit redundant to me. All other campaign behaviors aren't singletons. It'd make more sense to just construct a PartyAttributeManager here and add it to the CampaignBehaviors that way. No reason to make it any more global than we have to.

Reply:
you are right I removed the singleton property. This was fair enough. For a certain form in between it was necessary but I left it because I was lazy :D